keyhub/provider.go

```diff
@@ -92,3 +96,21 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}
 
 	return client, diags
 }
+
+func apiErrorToLogFields(err error) map[string]interface{} {
+
+	fields := map[string]interface{}{}
+
+	var apiError keyhubmodel.KeyhubApiError
+	if errors.As(err, &apiError) {
+		fields["code"] = fmt.Sprintf("%d", apiError.Report.Code)
+		fields["reason"] = apiError.Report.Reason
+		fields["exception"] = apiError.Report.Exception
+		fields["message"] = apiError.Report.Message
+		fields["applicationError"] = apiError.Report.ApplicationError
+		fields["stacktrace"] = strings.Join(apiError.Report.StackTrace, "\n")
+	}
```

Comment:
Niet al deze velden zijn altijd aanwezig. Ik verwacht dat Go daar niet goed mee omgaat. Check in de if in ieder geval of `apiError.Report` wel bestaat (dat zal meestal wel het geval zijn, maar als bijv de proxy een fout geeft, dan niet). En bij de `stacktrace` verwacht ik ook problemen, als die ontbreekt.